lib/rucio/daemons/rsedecommissioner/profiles/atlas.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""ATLAS-specific decommissioning profiles."""
+
+import logging
+from typing import TYPE_CHECKING
+
+from rucio.core.rule import list_rules_for_rse_decommissioning
+
+from .generic import (ActionDecision, generic_profile_template_function,
+                      _make_move_function, _generic_decide_action,
+                      _generic_finalize, _generic_initialize)
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from typing import Any
+
+
+def atlas_move(
+    rse: 'dict[str, Any]',
+    config: 'dict[str, Any]',
+    *,
+    logger: 'Callable[..., None]' = logging.log
+) -> None:
+    """Move rules that satisfy conditions to a specific destination.
+
+    The "ATLAS move" profile lists out all rules that are locking replicas
+    at the given RSE, and moves them to the specified destination if either
+    one of the following is true:
+
+    - The RSE expression of the rule is trivial (the RSE name itself).
+    - There are no replicas locked by the rule that reside on another RSE.
+    - The datatype of the DID is not "log".
+
+    :param rse: RSE table entry as a dictionary.
+    :param config: Decommissioning configuration dictionary.
+    :param logger: Logging function.
+    """
+    generic_profile_template_function(rse, config,
+                                      _generic_initialize,
+                                      list_rules_for_rse_decommissioning,
+                                      _atlas_decide_action,
+                                      _make_move_function(config['destination']),
+                                      _generic_finalize,
+                                      logger=logger)
+
+
+def _atlas_decide_action(
+    rule: 'dict[str, Any]',
+    rse: 'dict[str, Any]',
+    *,
+    logger: 'Callable[..., None]' = logging.log,
+) -> 'ActionDecision':
+    """Determine what to do with the given rule (ATLAS specific).
+
+    This function checks the DID metadata "datatype" and returns DONT_REMOVE
+    if the datatype is "log". Otherwise the decision follows the generic profile.
+
+    :param rule: Rule to delete.
+    :param rse: RSE table entry as a dictionary.
+    :param logger: Logging function.
+    :returns: Action flag for the rule.
+    """
+    # Get the DID metadata and check the datatype
```

Comment:
I could not figure out how to get the datatype of a DID given a rule (without issuing a database query).

Reply:
No, this information is not stored in the rule. Given a `scope` and `name`, you need to use `rucio.core.did.get_metadata()` (return a dictionary which should contain `datatype`).

Reply:
Thank you. Using get_metadata() now.